addon/chrome/content/preferences/preferences.js

```diff
@@ -52,13 +52,29 @@ var gPrefWindow = {
     window.addEventListener("input", this);
 
     window.document.querySelectorAll('input[type=number]').forEach(item => {
+      item.addEventListener("keypress", i => {
+        if (i.charCode == 0) return;
+        if (!(i.altKey || i.ctrlKey || i.metaKey) &&
+          (!(i.key == '-' &&
+            (i.target.editor.selection.anchorOffset == 0 ||
+              i.target.editor.selection.focusOffset == 0) ||
+            (i.charCode >= 48 && i.charCode <= 57)) ||
+            i.target.editor.textLength > i.target.maxLength ||
+            (i.target.editor.textLength == i.target.maxLength &&
+              i.target.editor.selection.anchorOffset == i.target.editor.selection.focusOffset))) {
+          i.stopPropagation();
+          i.preventDefault();
+        }
+      }, true);
       item.addEventListener("change", i => {
-        if(!i.target.validity.valid) item.value = item.defaultValue
+        if (!i.target.validity.valid) item.value = item.defaultValue
       });
       item.addEventListener("input", i => {
-        if(this.instantApply || !i.target.validity.valid) i.stopPropagation();
-        if (i.target.value.length>i.target.maxLength) item.value = item.defaultValue;
-        else if(i.target.validity.valid) item.defaultValue = item.value;
+        if (this.instantApply || !i.target.validity.valid) i.stopPropagation();
+        if (i.target.editor.textLength > i.target.maxLength || (i.target.validity.badInput &&
+          i.target.editor.selection.anchorNode.wholeText != "-"))
+          item.value = item.defaultValue;
+        else if (i.target.validity.valid) item.defaultValue = item.value;
```

Comment:
I think you can remove this line, it is the default

Reply:
This is for capture paste in contents. And "number" type input won't take 'maxlength' for default.

Reply:
just line 77

Reply:
That is for capture last valid input.